### PR TITLE
Refactor to ES6

### PR DESCRIPTION
**Pull request by fluxsauce**
Howdy,

As requested in https://github.com/pbojinov/request-ip/issues/22 , I've refactored to ES6.

What changed:

- Added `.editorconfig` that matches project's existing style - see http://editorconfig.org for details
- Added ESLint standards (extending [airbnb-base](https://github.com/airbnb/javascript))
- Updated major version to `2.0.0` (ES6 vs the ES5 `1.x` releases)
- Switched `istanbul` for successor, `nyc` - see https://github.com/istanbuljs/nyc for details
- Added dependency to [is.js](http://is.js.org/), a micro check library for IP and other validation validation

Code coverage is at 84%.

Let me know if you have any questions, comments or concerns, happy to help and discuss!

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,149 +1,136 @@
...
+        req[attributeName] = getClientIp(req); // eslint-disable-line no-param-reassign
         next();
-    }
-};
\ No newline at end of file
+    };
+}
+
+module.exports = {
+    getClientIpFromXForwardedFor,
```
Comment (pbojinov) [inline index.js]: I don't think we need to export `getClientIpFromXForwardedFor`. Should be a private method for `getClientIp`.

**Comment by pbojinov** [comment]
Very nice! Love the added JSDocs and ESLint. I'll review it

Any idea why the coverage dropped from 100% to 85%, just curious?

**Comment by fluxsauce** [comment]
> Any idea why the coverage dropped from 100% to 85%, just curious?

It's not making it to the of the block; I removed some logic that made some test-only changes to the headers. A better way would be to just mock it rather than making a request, which I can work on.

**Comment by pbojinov** [comment]
That makes sense if its not making it to some of the test only logic.

That'd be really cool to see a mock POC, it would definitely make the tests run much faster. 

We could leave the request tests as the integration tests and the mocks could the quick sanity checks.

File: index.js (hunk 1 above)
Comment (fluxsauce) [inline index.js]: Keeping as it makes it explicitly testable on its own.

**Comment by fluxsauce** [comment]
@pbojinov Updated!

File: test/index.js [hunk 2]
```diff
@@ -326,6 +327,56 @@ test('req.connection.socket.remoteAddress', (t) => {
     });
 });
 
+test('getClientIp - req.connection.remoteAddress', (t) => {
+    t.plan(1);
+    const found = requestIp.getClientIp({
```
Comment (pbojinov) [inline test/index.js]: genius!

File: index.js (hunk 1 above)
Comment (pbojinov) [inline index.js]: Makes sense, anything in the name of testabliity 👍

**Pull request merged**